### DISCUSSION TITLE
Proposal for Waterfall chart improvement in handling first and last subtotals (do not add SIP number)

### DISCUSSION THREAD

**Discussion by code-luac** [apache/superset#31441] [Ideas]
*Please make sure you are familiar with the SIP process documented*
[here](https://github.com/apache/superset/issues/5602). The SIP will be numbered by a committer upon acceptance.

### Motivation

1. Today, the implementation assumes that the first value in the series is zero. However, this is not always the case. For example, I want to review the evolution of my task backlog over time (opened vs. closed tickets and subtotals with a weekly time grain). When filtering for a particular time window (not from the beginning of time), I want to see how many open tickets there were at the beginning of this period.
2. Today, the last bar in the chart displays the subtotal. I want to control whether this last subtotal appears or not.

This brings the waterfall chart to par with Google Sheets and MS Excel.

### Proposed Change
I used the following screenshot from Google Sheets to serve as my wireframe.
Requirements for the Waterfall Chart settings (see the markup in the screenshot):
1.  A new checkbox "Use first value as subtotal" - when checked, render the first bar in the chart as a subtotal (gray) using the first value in the series.
2. A new checkbox "Add subtotal after last value in series" - when checked, render the last bar in the chart as a subtotal (gray) representing the series' total.
![image](https://github.com/user-attachments/assets/6eb681e3-5808-4b64-bf65-f8bf43580409)

### New or Changed Public Interfaces

Describe any new additions to the model, views or `REST` endpoints:
Change the chart settings API to reflect the 2 new settings checkboxes.

Describe any changes to existing visualizations, dashboards and React components. Describe changes that affect the Superset CLI and how Superset is deployed:
See proposed change section above.

### New dependencies

None

### Migration Plan and Compatibility

None

### Rejected Alternatives

None

**Comment by michael-s-molina** [comment]
Hi @code-luac. I believe this is a valid feature request that does not require a SIP given its scope. I don't have the bandwidth to implement this now as I'm focused on stabilizing 4.1 but I hope someone from the community can provide such feature as it would be valuable.

**Comment by rusackas** [comment]
Agreed with Michael. We'd welcome a PR with these changes. If you don't want to consider it a SIP, I think there are two potential moves for this issue:
1) We can move it to an Ideas discussion thread if you want more discussion/feedback/help with it
2) We can leave it here if you have more immediate plans to open a PR that closes it. 
Let us know which seems appropriate, and thanks for the input/ideas!
